cpp/src/arrow/flight/sql/odbc/odbcabstraction/include/odbcabstraction/odbc_impl/encoding_utils.h

```diff
@@ -77,4 +77,37 @@ inline size_t ConvertToSqlWChar(const std::string& str, SQLWCHAR* buffer,
   }
 }
 
+/// @brief Convert buffer of SqlWchar to standard string
+/// @param wchar_msg[in] SqlWchar to convert
+/// @param msg_len[in] Number of characters in wchar_msg
+/// @return wchar_msg in std::string format
+inline std::string SqlWcharToString(SQLWCHAR* wchar_msg, SQLSMALLINT msg_len = SQL_NTS) {
+  if (wchar_msg == nullptr) {
+    return std::string();
```

Comment:
Are there cases where an empty string should behave differently from a null string?